Source/CarthageKit/Project.swift

```diff
@@ -171,6 +181,38 @@ public final class Project {
 	private var cachedVersions: CachedVersions = [:]
 	private let cachedVersionsQueue = ProducerQueue(name: "org.carthage.CarthageKit.Project.cachedVersionsQueue")
 
+	/// Attempts to determine the local version of swift
+	private static func determineSwiftVersion() -> SignalProducer<String, CarthageError> {
+		let taskDescription = Task("/usr/bin/env", arguments: ["xcrun", "swift", "--version"])
+
+		return taskDescription.launch(standardInput: nil)
+			.ignoreTaskData()
+			.mapError(CarthageError.taskError)
+			.map { data -> String? in
+				return parseSwiftVersionCommand(output: String(data: data, encoding: .utf8))
+			}
+			.flatMap(.concat) { versionString -> SignalProducer<String, CarthageError> in
+				if let versionString = versionString {
+					return SignalProducer(value: versionString)
+				} else {
+					return SignalProducer(error: .unknownLocalSwiftVersion)
+				}
+			}
+	}
```

Comment:
Can you move this to `Xcode.swift`? It's really an Xcode thing, not Carthage-specific project logic.